### PR TITLE
Initial template schema, MariaDB for docker and general cleanup of mocks

### LINKED ISSUE #33: Define data model for Templates
Need to define the schema behind the initial Template builder [wireframes](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=69-345&t=N2jvNbe3g9eOxDc7-0&scaling=min-zoom&page-id=6%3A66&starting-point-node-id=69%3A345) along with any queries and mutations needed to interact with the data.

Also make note of any logic or questions that pop up

### PR DESCRIPTION
**Pull request by briri**
## Description

Initial schema, model, resolver (queries only), database tables and seed data for a Template. This PR is for #14 and based off of initial definition in #33

All SQL logic lives in the Models now, with Resolvers simply making calls to the model to perform those queries. In the future, the Resolvers will perform authorization checks and business logic (e.g. check that the user has permission, generate a new version of the template, save the new version, notify stakeholders, etc.)

Additions:
- Replaced the old ApolloServerContext mock with a new one that is more accurate and usable
- Added `MySqlModel` abstract class that manages common properties and functions
- Added `Collaborator` abstract class and `TemplateCollaborator` class (models, schema, and resolvers)
- Added `Template` class (model, schema, and resolver)
- Added `VersionedTemplate` class (model, schema, and resolver)
- Added `TemplateService` that performs `generateVersion` and `clone`

Updates:
- Move Affiliation API calls from the Resolver to the Model 
- Updated `ContributorRole` and `User` models, resolvers and schema to be like the new models
- Updated the `src/context.ts` to have a callable `buildContext()` function so that it is usable through the normal Apollo server context as well as within our Express endpoints for `login` and `register`
- Changed some `private` properties in classes to `protected` because they were giving me grief in Jest mocks. (Unable to mock a private property/function)
- Removed the `Model` suffix on some of the model classes so that names are consistent
- Removed old `src/config/AwsConfig.ts`. We weren't using it and when we do need to access AWS resources, aws-sdk v3 is different
- Removed all of the old Apollo server mocks (except Affiliation ... left that as an example in case we want to revisit) since we could not get that working and all of the tables have been implemented.
- Updated all other `data-migration/*.sql` files so that the tables all have the standard: `created`, `modified`, `id`, `createdById` and `modifiedById` fields.
- Reordered a few of the `data-migration/` SQL files so that they run in a correct order so the DB can be built from scratch

Other bits:
- Fixed the ESLint config and added it as a husky pre-commit task
- Updated the README with accurate info
- removed old MySQL config from `.env.example` (
- Added MariaDB to docker-compose
- Added script to `data-migrations/` that drops all of the tables in the local DB

### Testing
- Update your local `.env` and remove any references to MySQL
- startup the container: `docker-compose up`
- Nuke your existing DB: `docker-compose exec apollo bash ./data-migrations/nuke-db`
- Rebuild the DB with new tables and seed data: `docker-compose exec apollo bash ./data-migrations/process.sh`
- Visit the GraphQL explorer `http://localhost:4000/graphql` and check out the new Queries

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
     "compile": "tsc",
     "copy-graphql": "mkdir ./dist/schemas && cp ./src/schemas/*.graphql ./dist/schemas/",
-    "build": "rm -rf ./dist && npm run compile",
     "generate": "graphql-codegen",
-    "dev": "ts-node-dev --respawn ./src/index.ts",
+    "lint": "npx eslint . --ignore-pattern dist/ --ignore-pattern src/types.ts",
     "test": "jest --coverage",
-    "test-tsc": "jest -c ./jest.tsc.config.js",
-    "lint": "npx eslint .",
+    "build": "rm -rf ./dist && npm run lint && npm run generate && npm run test && npm run compile",
```
Comment (jupiter007) [inline package.json:12]: @briri - I'm wondering if we should exclude "`npm run lint`" and "`npm run test`" from "`build`". In the code pipeline, we will probably only want to run "build" after the linting and tests pass.

File: README.md [hunk 2]
```diff
@@ -9,19 +9,53 @@ Our Apollo server installation consists of:
...
-    - If you have homebrew installed, run `brew install mysql` and then start it `brew services start mysql`
-  - Initialize the Database and the dataMigrations table: `./data-migrations/database-init.sh`
-  - Run all database migrations: `./data-migrations/process.sh`
-- Install all of the dependencies: `npm install`
-- Generate the Typescript types: `npm run generate`
-- Startup the application in development mode: `npm run dev`
-- Navigate to `http://localhost:4000` in your browser
+- Clone this repository to your local machine
+- Run `docker-compose build` to build the container
+- In a separate tab run `docker-compose exec apollo bash ./data-migrations/process.sh` to build out the local database
```
[the remark this answers is not in the document]
Comment (briri) [reply README.md:14]: Ahh right. You need to have the container running in order for that to work. I will update these instructions

File: data-migrations/2024-05-01-1103-create-user-table.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE `users` (
```
[the remark this answers is not in the document]
Comment (briri) [reply data-migrations/2024-05-01-1103-create-user-table.sql:1]: I considered that, but I think I would prefer to do in a separate ticket when we update the schema to support them.

File: package.json (hunk 1 above)
Comment (briri) [reply package.json:12]: Yes. You are right. I will remove

File: tsconfig.json [hunk 4]
```diff
@@ -9,4 +9,5 @@
   },
   "include": ["src", "src/__tests__"],
   "exclude": ["node_modules"],
+  "files": ["node_modules/jest-expect-message/types/index.d.ts"],
```
[the remark this answers is not in the document]
Comment (briri) [reply tsconfig.json:12]: Yes it seems to have. I am sure I copied the above from somewhere 🙃

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (briri) [reply README.md:17]: @jupiter007  I added some additional info to the installation instructions in the README about setting up the .env file

**Review by jupiter007 — APPROVED** [review]

**Pull request merged**